### PR TITLE
fix: [CI-4654]: Use existing delegate api to fetch status

### PR DESCRIPTION
**Pull request by vardanbansal-harness**
### Summary

As part of this change, we'll now be calling "Delegate group by identifier" api `/setup/delegates/ng/v2/{identifier}`, every 20 seconds to obtain delegate installation status.

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-ci/hooks/useProvisionDelegateForHostedBuilds.ts [hunk 1]
```diff
@@ -48,16 +52,15 @@ export function useProvisionDelegateForHostedBuilds(): ProvisionDelegateForHoste
   })
 
   useEffect(() => {
-    const { status, data } = provisioningStatus || {}
-    if (status === Status.SUCCESS && data === ProvisioningStatus[ProvisioningStatus.SUCCESS]) {
+    if (!fetchingDelegateDetails && delegateDetails?.resource && delegateDetails.resource?.activelyConnected) {
```
Comment (d1wilko) [inline src/modules/75-ci/hooks/useProvisionDelegateForHostedBuilds.ts:55]: I think this just needs to be `if (!fetchingDelegateDetails && delegateDetails?.resource?.activelyConnected)`

**Review by d1wilko — CHANGES_REQUESTED** [review]
just one change - but otherwise looks good to go

File: src/modules/75-ci/hooks/useProvisionDelegateForHostedBuilds.ts (hunk 1 above)
Comment (vardanbansal-harness) [reply src/modules/75-ci/hooks/useProvisionDelegateForHostedBuilds.ts]: Done.

**Review by d1wilko — APPROVED** [review]
LGTM

**Review by d1wilko — APPROVED** [review]

**Pull request merged**